packages/common/src/sites/_internal/SiteSchema.ts

```diff
@@ -17,11 +17,17 @@ export const SiteEditorTypes = [
 /**
  * defines the JSON schema for a Hub Site's editable fields
  */
-export const SiteSchema: IConfigurationSchema = {
-  ...HubItemEntitySchema,
-  properties: {
-    ...HubItemEntitySchema.properties,
-    _discussions: ENTITY_IS_DISCUSSABLE_SCHEMA,
-    telemetry: PRIVACY_CONFIG_SCHEMA,
-  },
-} as IConfigurationSchema;
+export const getSiteSchema = (siteId: string) =>
+  ({
+    $async: true,
+    ...HubItemEntitySchema,
+    properties: {
+      ...HubItemEntitySchema.properties,
+      _discussions: ENTITY_IS_DISCUSSABLE_SCHEMA,
+      telemetry: PRIVACY_CONFIG_SCHEMA,
+      _urlInfo: {
+        type: "object",
+        isUniqueDomain: { siteId },
```

Comment:
I'm far from a json schema expert, but how does this work when creating a new site, when there is no siteId?

Reply:
that's a good question.

short answer: it's fine

long answer: the keyword validator that this schema is passed into only uses the `siteId` if it was passed in. Here's the logic:

```ts
  // siteId is only passed in when editing an existing site
  const siteId = schema?.siteId;
  let isValid;
  try {
    const domain = await lookupDomain(hostname, context.hubRequestOptions);
    // if lookupDomain does NOT throw, we found a domain record
    // which is only valid if it is the one associated with the site we are editing
    isValid = !!siteId
      ? domain.siteId === siteId
      : false;
  } catch (_) {
    // no domain found, so it's available
    isValid = true;
  }
  return isValid;
```